### PR TITLE
Issue 249: added topic containing information on the S2I build process

### PR DESCRIPTION
**Pull request by inoxx03**
@rhoads-zach Mind taking a look? Also, where in the docs do we include this?

### CODE REVIEW

**Comment by inoxx03** [comment]
Fixes #249

**Comment by rhoads-zach** [comment]
@inoxx03 can you make this topic appendix A in the 3 dev guides:

https://github.com/openshiftio/appdev-documentation/blob/master/docs/vertx-runtime/master.adoc
https://github.com/openshiftio/appdev-documentation/blob/master/docs/wf-swarm-runtime/master.adoc
https://github.com/openshiftio/appdev-documentation/blob/master/docs/spring-boot-runtime/master.adoc

File: docs/topics/topic-s2i-build-process.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+With S2I builds you can  easily deliver the latest version of your application into production while achieving shorter build times, decreased resource and network usage, improved security and a number of link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[additional advantages].
+
+You need to provide three key elements as the input to the S2I process:
+
+* application sources (hosted in an online SCM repository, such as a Git Hub repo),
+
+* link:https://docs.openshift.com/container-platform/3.4/creating_images/s2i.html#s2i-scripts[S2I scripts],
+
+* the Builder image, which serves as the foundation for the assembled image and provides the ecosystem within which you application is running.
+
```
Comment (rhoads-zach) [inline docs/topics/topic-s2i-build-process.adoc]: In the above list, you need to either remove the punctuation at the end, or turn all 3 in to sentences and end them with periods

File: docs/topics/topic-s2i-build-process.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+The process combines these inputs by injecting your application sources and  dependencies into the Builder image according to instructions specified in the S2I script, and generates a Docker-formatted container image that runs the assembled application. For more detail, check the
```
Comment (rhoads-zach) [inline docs/topics/topic-s2i-build-process.adoc]: remove double space `and  dependencies`

Comment (rhoads-zach) [inline docs/topics/topic-s2i-build-process.adoc]: No comma here:
> script, and generates

File: docs/topics/appendix-s2i-build-process.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+You need to provide three key elements as the input to the S2I process:
+
+* application sources (hosted in an online SCM repository, such as a Git Hub repo),
+
+* link:https://docs.openshift.com/container-platform/3.4/creating_images/s2i.html#s2i-scripts[S2I scripts],
+
+* the Builder image, which serves as the foundation for the assembled image and provides the ecosystem within which you application is running.
+
+The process combines these inputs by injecting your application sources and  dependencies into the Builder image according to instructions specified in the S2I script, and generates a Docker-formatted container image that runs the assembled application. For more detail, check the
+link:https://docs.openshift.com/container-platform/3.4/creating_images/s2i.html[S2I build requirements] and link:https://docs.openshift.com/container-platform/3.4/welcome/index.html[build options] sections of the link:https://docs.openshift.com/container-platform/3.4/welcome/index.html[OpenShift Container Platform documentation].
```
Comment (rhoads-zach) [inline docs/topics/appendix-s2i-build-process.adoc]: The last two links are the same

File: docs/topics/topic-s2i-build-process.adoc (hunk 1 above)
Comment (inoxx03) [inline docs/topics/topic-s2i-build-process.adoc]: removed the punctuation from the list items

File: docs/topics/topic-s2i-build-process.adoc [hunk 4]
```diff
@@ -0,0 +1,15 @@
+= The Source-to-Image (S2I) Build Process
+
+link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image] is a build tool that allows you to generate reproducible Docker-formatted container images from application sources hosted in an online SCM repository.
+With S2I builds you can  easily deliver the latest version of your application into production while achieving shorter build times, decreased resource and network usage, improved security and a number of link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[additional advantages].
```
Comment (inoxx03) [inline docs/topics/topic-s2i-build-process.adoc]: Changed

File: docs/topics/appendix-s2i-build-process.adoc (hunk 3 above)
Comment (inoxx03) [inline docs/topics/appendix-s2i-build-process.adoc]: Thanks. Fixed.

File: docs/topics/topic-s2i-build-process.adoc (hunk 2 above)
Comment (inoxx03) [inline docs/topics/topic-s2i-build-process.adoc]: thanks:) I did not notice the extra comma.

**Comment by rhoads-zach** [comment]
merged: https://github.com/openshiftio/appdev-documentation/commit/163a7870f5328e2b964ad5d281ef4284bca42edd

File: docs/topics/appendix-s2i-build-process.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
+//= The Source-to-Image (S2I) Build Process
+link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image] (S2I) is a build tool that allows you to generate reproducible Docker-formatted container images from application sources hosted in an online SCM repository.
+With S2I builds you can easily deliver the latest version of your application into production while achieving shorter build times, decreased resource and network usage, improved security and a number of link:https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/builds_and_image_streams.html#source-build[advantages].
+
+You need to provide three key elements as the input to the S2I process:
+
+* application sources (hosted in an online SCM repository, such as a Git Hub repo)
```
Comment (Ladicek) [inline docs/topics/appendix-s2i-build-process.adoc]: It's actually GitHub, without the space.

**Pull request closed without merging**